### PR TITLE
Column Customization: Refactor Column Headers to include enabled

### PR DESCRIPTION
**Pull request by JamesHollyer**
* Motivation for features / changes
Our goal is to allow users to select which columns they what displayed in the data table and in which order. To do this we have to have a notion of which columns are enabled and which are disabled. This change makes the column headers in the ngrx state objects with have a type and an enabled boolean so that this info can be stored and utilized.

### CODE REVIEW

File: tensorboard/webapp/widgets/data_table/data_table_component.ts [hunk 1]
```diff
@@ -281,4 +295,11 @@ export class DataTableComponent implements OnDestroy {
       'highlight-border-left': this.highlightSide === Side.LEFT,
     };
   }
+
+  showColumn(header: ColumnHeader) {
+    return (
+      header.enabled &&
+      (this.smoothingEnabled || header.type !== ColumnHeaderType.SMOOTHED)
```
Comment (rileyajones) [inline tensorboard/webapp/widgets/data_table/data_table_component.ts:302]: Could we just "disable" the SMOOTHED column when smoothing is disabled and avoid this check? It would be fine to make this change in another PR btw.

File: tensorboard/webapp/widgets/data_table/data_table_component.ts [hunk 2]
```diff
@@ -223,41 +228,50 @@ export class DataTableComponent implements OnDestroy {
...
+    if (!this.draggingHeaderType || !this.highlightedColumnType) {
       return;
     }
 
     this.orderColumns.emit(
       this.moveHeader(
-        this.headers.indexOf(this.draggingHeader),
-        this.headers.indexOf(this.highlightedColumn)
+        this.headers.findIndex((element) =>
+          isHeaderOfType(this.draggingHeaderType!, element)
```
Comment (rileyajones) [inline tensorboard/webapp/widgets/data_table/data_table_component.ts:246]: This function seems unnecessarily complicated. Could you either wrap the entire call with something along the lines of `getIndexOfDraggingHeader` or else remove the `isHeaderOfType` function?

**Review by rileyajones — APPROVED** [review]
Mostly LGTM.
In the future it would be nice to have the renames split apart from the added logic.

File: tensorboard/webapp/widgets/data_table/data_table_component.ts (hunk 1 above)
Comment (JamesHollyer) [reply tensorboard/webapp/widgets/data_table/data_table_component.ts:290]: The Smoothed column being disabled is different from it not being shown. When disabled it will not appear when the smoothing factor is set.  However, as we discussed offline we plan to leave the SMOOTHED column in even when smoothing is not enabled. This change will come in a future PR.

File: tensorboard/webapp/widgets/data_table/data_table_component.ts (hunk 2 above)
Comment (JamesHollyer) [reply tensorboard/webapp/widgets/data_table/data_table_component.ts]: Good suggestion. Done.
